core/src/main/scala/kafka/server/AlterIsrManager.scala

```diff
@@ -68,28 +67,14 @@ object AlterIsrManager {
    * Factory to AlterIsr based implementation, used when IBP >= 2.7-IV2
    */
   def apply(
-    config: KafkaConfig,
-    metadataCache: MetadataCache,
+    alterIsrChannelManager: BrokerToControllerChannelManager,
     scheduler: KafkaScheduler,
     time: Time,
-    metrics: Metrics,
-    threadNamePrefix: Option[String],
     brokerEpochSupplier: () => Long,
     brokerId: Int
   ): AlterIsrManager = {
-    val nodeProvider = MetadataCacheControllerNodeProvider(config, metadataCache)
```

Comment:
Are these changes just cleaning up? Did we not want to add an extra parameter to `DefaultAlterIsrManager`?

Reply:
Yes, this is just a cleanup. 
Currently, we have several `BrokerToControllerChannel`  for AlterIsrManager, LifecycleManager, AutoTopicCreateManager and ForwardingManager. My cleanup here has 2 benifits:
1. Create and manage them all in `BrokerServer` or `KafkaServer`
2. Avoid creating different `ControllerNodeProvider` for each of them